### PR TITLE
Development

### PR DESCRIPTION
**Pull request by AnthonyGW**
#### What does this PR do?
Add Create, Read and Delete functionality in React JS that uses the remote Bucketlist API endpoints
#### Description of Task to be completed?
Make use of the following API endpoints:
`/auth/register`
`/auth/login`
`/auth/logout`
`/auth/reset-password`
`/bucketlists/`
`/bucketlists/<int: id>`
`/bucketlists/<int: list_id>/items`
`/bucketlists/<int: list_id>/items/<int: item_id>`
#### How should this be manually tested?
There are no test files implemented yet.
Test functionality by interacting with the GUI forms and buttons, and UI paths.
#### Any background context you want to provide?
The functionality is based on the challenge 2 bootcamp project which can be found here: [BucketListUltima](https://github.com/AnthonyWaithaka/Bucket-List-Ultima)

Features/Tests to be implemented next:
1. Reset user password
2. ~~Update bucketlist info~~
3. ~~Update bucketlist item info~~
4. UI/UX design with ReactBootstrap
5. Results pagination
6. Search feature
7. Component testing with Enzyme or Jest
8. Feature testing with Selenium-based libraries
9. Continuous Integration with TravisCI, Test coverage reporting with Coveralls, CI badges

### CODE REVIEW

**Comment by AnthonyGW** [comment]
I've added the features for updating bucketlists and bucketlist items. Also crossed them out on the list

**Review by jonathankamau — CHANGES_REQUESTED** [review]
Please run this command: `git pull --rebase master` on your development branch. You have too many unnecessary files on your PR.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+**.cache/
```
Comment (jonathankamau) [inline .gitignore:1]: A new line is required here

**Review by jonathankamau — COMMENTED** [review]
I am unable to view your code because of the many files. Run the `git pull --rebase master` to remove the unnecessary files from your PR.

**Pull request closed without merging**